about_matching.exs

```diff
@@ -33,6 +33,20 @@ defmodule AboutMatching do
     assert_raise MatchError, fn -> [a , a] = [1 , __?] end
   end
 
+  think "Matched values overwrite variable values" do
+    a = 1
+    assert a == __?
+
+    [a, _] = [2, 3]
+    assert a == __?
+  end
+
+  think "Values can be pinned to prevent them from being overwritten" do
```

Comment:
The more I think about this, I'm leaning back to how it's currently written. The "meets your expectation" is more _why_ you would pin a variable. The example describes the actual behavior. I wonder if there's a way to work both of those into the description...